src/mapping.ts

```diff
@@ -0,0 +1,275 @@
+  color = "WHITE";
+  log.info("isAlpha: {}", [check.toString()]);
+
+  if (check) {
+    color = "BLACK";
+    return color;
+  } else if (_tier == "ONE") {
+    return color;
+  } else if (_tier == "TWO") {
+    color = "YELLOW";
+  } else if (_tier == "THREE") {
+    color = "BLUE";
+  } else if (_tier == "FOUR") {
+    color = "RED";
+  }
+  return color;
+}
+
+/**
+ * @notice sort tier as a string to fit schema enum reqs
+ * @param _tier
+ * @returns tier: string
+ */
+function checkMergeClass(_tier: BigInt): string {
+  let tier: string;
+  tier = "ONE";
+  if (_tier == BigInt.fromI32(1)) {
+    return tier;
+  } else if (_tier == BigInt.fromI32(2)) {
+    tier = "TWO";
+  } else if (_tier == BigInt.fromI32(3)) {
+    tier = "THREE";
+  } else if (_tier == BigInt.fromI32(4)) {
+    tier = "FOUR";
+  }
+
+  return tier;
+}
+
+// /**
+//  * @notice object type where properties represent value-based-decoded nft metadata
+//  * TODO: not sure but I am getting a compile error here. Asked DK. I need to just look into this. This will be done in next PR as I clean up the code.
+//  */
+// type Values = {
+//   massSize: BigInt;
+//   tier: string;
+//   color: string;
+// };
+
+// /**
+//  * @notice carries out decoding of value for a particular mass NFT
+//  * @returns value : Values
+//  * @dev TODO: see Object Type Values above and the error that is there. Once that is resolved then this can be used.
+//  */
+// function decodeValue(nftValue: BigInt): Values {
+//   let value: Values;
+//   let _tier: BigInt = nftValue / CLASS_MULTIPLIER;
+//   let tier = checkMergeClass(_tier);
+
+//   value = {
+//     massSize: (nftValue % CLASS_MULTIPLIER) as BigInt,
+//     tier: tier,
+//     color: checkColor(tier),
+//   };
+
+//   return value;
+// }
+
+/* ========== EVENT HANDLERS ========== */
+
+/**
+ * @notice handles Transfer events and ultimately instantiates / updates User and NFT entities
+ * @param event: Transfer
+ * @dev all NFT entity generation is handled in other functions.
+ */
+export function handleTransfer(event: Transfer): void {
+  const to = event.params.to.toHex();
+  const from = event.params.from.toHex();
+
+  let nft: NFT;
+
+  let user = User.load(to);
+  const tokenId = event.params.tokenId.toString();
+
+  // Step 0: set NG_OMNIBUS as a whitelisted address as it is done in the constructor
+  if (!user && to == NG_OMNIBUS) {
+    user = new User(to);
+    user.whitelist = true;
+  }
+
+  // Step 1: checks if this is a new user, takes care of first time NG mints too.
+  if (!user) {
+    user = new User(to);
+  } // checks if this is a mint() scenario
+
+  user.save();
+
+  // Step 2a: either create new NFT if in minting phase
+  if (from == EMPTY_ADDRESS && to == NG_OMNIBUS) {
+    nft = createNFT(event, user.id);
+  }
+
+  // Step 2b: OR updateNFT for all other Transfer scenarios incl: temporary transfers right before merging/burning, simple burning (direct to dead address, which then transfers to address(0))
+  else {
```

Comment:
please put the `else` right after the ending bracket of the `if` right before it. this is standard practice. the reason your linter is not catching it is because you have a big comment in between them

Reply:
i.e.:
```js
  if (from == EMPTY_ADDRESS && to == NG_OMNIBUS) {
    nft = createNFT(event, user.id);
  } else {
    nft = updateNFT(tokenId, user.id);
  }
  ```

Reply:
Updated.